Content.Server/StationReport/StationReportSystem.cs

```diff
@@ -0,0 +1,47 @@
+using Content.Server.GameTicking;
+using Content.Shared.StationReport;
+using Content.Shared.Paper;
+using Robust.Shared.GameObjects;
+
+namespace Content.Server.StationReportSystem;
+
+public sealed class StationReportSystem : EntitySystem
+{
+
+    //this is shitcode
+    public string? StationReportText;
+
+    public override void Initialize()
+    {
+        //sets stationreporttext to null so it doesn't keep from the previous round
+        base.Initialize();
+        if (StationReportText != null)
+        {
+            StationReportText = null;
+        }
+        //subscribes to the endroundevent
+        SubscribeLocalEvent<RoundEndTextAppendEvent>(OnRoundEndTextAppend);
+    }
+
+    private void OnRoundEndTextAppend(RoundEndTextAppendEvent args)
+    {
+        //locates the first entity with StationReportTabletComponent then stops
+        var query = EntityQueryEnumerator<StationReportTabletComponent>();
+        while (query.MoveNext(out var uid, out var tablet))
+        {
+            if (EntityManager.TryGetComponent<PaperComponent>(uid, out var paper))
+            {
+                StationReportText = paper.Content;
+                break;
+            }
+        }
+        BroadcastStationReport(StationReportText);
+    }
```

Comment:
Why do you need StationReportText? Just use the paper.Content directly.

You're running an EntityQueryEnumerator on the component of the entity that holds the text you want. You dont need StationReportText at all. Just make it a var inside here. Remove anything you did in Initialize().

Test the changes this before commiting btw.

```suggestion
        //locates the first entity with StationReportTabletComponent then stops
        var stationReportText;
        var query = EntityQueryEnumerator<StationReportTabletComponent>();
        while (query.MoveNext(out var uid, out var tablet))
        {
            if (!TryComp<PaperComponent>(uid, out var paper))
               return;
            
            stationReportText = paper.Content;
            break;
        }
        BroadcastStationReport(stationReportText);
    }
```

Reply:
Alr I got it working, just had to delete some stuff and make the variable nullable, thank for the better code brochacho